### PR TITLE
Fix --enable-zph-qos build [-Wunused-parameter]

### PR DESCRIPTION
**Pull request by rousskov**

### CODE REVIEW

**Comment by rousskov** [comment]
Disclaimer: I have not tested these (trivial) changes. I asked @elico (who reported the build problem) to test.

File: src/AclRegs.cc [hunk 1]
```diff
@@ -162,8 +162,8 @@ Acl::Init()
     RegisterMaker("transaction_initiator", [](TypeName name)->ACL* {return new TransactionInitiator(name);});
 
 #if USE_LIBNETFILTERCONNTRACK
-    RegisterMaker("clientside_mark", [](TypeName name)->ACL* { return new Acl::ConnMark; });
-    RegisterMaker("client_connection_mark", [](TypeName name)->ACL* { return new Acl::ConnMark; });
+    RegisterMaker("clientside_mark", [](TypeName)->ACL* { return new Acl::ConnMark; }); // XXX: Add name parameter to ctor
+    RegisterMaker("client_connection_mark", [](TypeName)->ACL* { return new Acl::ConnMark; }); // XXX: Add name parameter to ctor
```
Comment (rousskov) [inline src/AclRegs.cc:166]: I added those `XXX`s for consistency with other very similar lines in this file. If your think these marks should not be added, please remove them:

```suggestion
    RegisterMaker("clientside_mark", [](TypeName)->ACL* { return new Acl::ConnMark; });
    RegisterMaker("client_connection_mark", [](TypeName)->ACL* { return new Acl::ConnMark; });
```

**Comment by elico** [comment]
@rousskov I am testing this.
Will have an answer on the build matrix tomorrow.

**Comment by elico** [comment]
@rousskov The build matrix of CentOS, Oracle, Fedora, Amazon verified that the fix is working as expected.

**Comment by rousskov** [comment]
> The build matrix of CentOS, Oracle, Fedora, Amazon verified that the fix is working as expected.

Are they all ./configure Squid with `--enable-zph-qos`?

**Comment by elico** [comment]
No, There are all have the default RPMS build flags which are:

```

# squid -v

Squid Cache: Version 5.5

Service Name: squid

 

This binary uses OpenSSL 1.1.1k  FIPS 25 Mar 2021. For legal restrictions on distribution see https://www.openssl.org/source/license.html

 

configure options:  '--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/libexec' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--exec_prefix=/usr' '--libexecdir=/usr/lib64/squid' '--localstatedir=/var' '--datadir=/usr/share/squid' '--sysconfdir=/etc/squid' '--with-logdir=/var/log/squid' '--with-pidfile=/var/run/squid.pid' '--disable-dependency-tracking' '--enable-follow-x-forwarded-for' '--enable-auth' '--enable-auth-basic=DB,LDAP,NCSA,PAM,POP3,RADIUS,SASL,SMB,getpwnam,fake' '--enable-auth-ntlm=fake' '--enable-auth-digest=file,LDAP,eDirectory' '--enable-auth-negotiate=kerberos,wrapper' '--enable-external-acl-helpers=wbinfo_group,kerberos_ldap_group,LDAP_group,delayer,file_userip,SQL_session,unix_group,session,time_quota' '--enable-cache-digests' '--enable-cachemgr-hostname=localhost' '--enable-delay-pools' '--enable-epoll' '--enable-icap-client' '--enable-ident-lookups' '--enable-linux-netfilter' '--enable-removal-policies=heap,lru' '--enable-snmp' '--enable-storeio=aufs,diskd,ufs,rock' '--enable-wccpv2' '--enable-esi' '--enable-security-cert-generators' '--enable-security-cert-validators' '--enable-icmp' '--with-aio' '--with-default-user=squid' '--with-filedescriptors=16384' '--with-dl' '--with-openssl' '--enable-ssl-crtd' '--with-pthreads' '--with-included-ltdl' '--disable-arch-native' '--without-nettle' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'CFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' 'LDFLAGS=-Wl,-z,relro  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld' 'CXXFLAGS=-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig' --enable-ltdl-convenience

 

```

 

I will run a build matrix with it enabled later on.

 

Eliezer 

 

----

Eliezer Croitoru

NgTech, Tech Support

Mobile: +972-5-28704261

Email: ***@***.*** ***@***.***>

**Comment by rousskov** [comment]
>>> The build matrix of CentOS, Oracle, Fedora, Amazon verified that the fix is working as expected.

>> Are they all ./configure Squid with `--enable-zph-qos`?

> No, There are all have the default RPMS build flags

Then the build matrix probably has not tested the changes in this PR. The changes in this PR are all inside the `--enable-zph-qos` guards.

AFAICT, your "Tested for all RPM builds and seem to build" remark on [squid-dev](http://lists.squid-cache.org/pipermail/squid-dev/2022-May/009784.html) appears to confirm that this PR fixes the build [failure](http://lists.squid-cache.org/pipermail/squid-dev/2022-May/009782.html) it was meant to address.

**Comment by yadij** [comment]
Changing tag from waiting-for-committer since this PR has not passe the commit criteria. It is waiting for reviewer votes of 10-day timeout - whichever occurs first.

**Comment by rousskov** [comment]
> Changing tag from waiting-for-committer since this PR has not passed the commit criteria.

The S-waiting-for-committer label does not imply the PR can be merged immediately. The label only implies that the PR is waiting for some privileged action. In this particular case, the PR was waiting for several such actions, including verifying that somebody else has (properly) tested the changes before clearing the PR for merging. Additional reviews never hurt, of course, but this PR is not really waiting for them -- it can be merged without additional reviews.

Anubis will take care of basic commit criteria like voting, timeouts, and CI test failures. Humans, on the other hand, should worry about things that Anubis cannot understand or do, like checking out-of-band test results or clearing a PR for merging.

**Comment by rousskov** [comment]
> @yadij added S-waiting-for-more-reviewers

I am interpreting this action as "Amos requests more reviews" and, hence, will not clear this PR for merging (until there are more reviews).

**Pull request closed without merging**